rainier-core/src/main/scala/com/stripe/rainier/sampler/Ehmc.scala

```diff
@@ -1,44 +1,76 @@
 package com.stripe.rainier.sampler
 
 import scala.collection.mutable.ListBuffer
+import Log._
+import java.util.concurrent.TimeUnit._
 
 /**
   * Empirical HMC - automated tuning of step size and number of leapfrog steps
   * @param l0 the initial number of leapfrog steps to use during the dual averaging phase
   * @param k the number of iterations to use when determining the
   * empirical distribution of the total number of leapfrog steps until a u-turn
   */
-final case class EHMC(l0: Int, k: Int = 2000) extends Sampler {
+final case class EHMC(l0: Int, k: Int) extends Sampler {
+
   def sample(density: DensityFunction,
              warmupIterations: Int,
              iterations: Int,
              keepEvery: Int)(implicit rng: RNG): List[Array[Double]] = {
 
     val lf = LeapFrog(density)
     val params = lf.initialize
+
+    FINE.log("Finding reasonable initial step size")
```

Comment:
we've pulled a lot more of the logic into this class and added logging. This becomes somewhat duplicative of HMC, but means it can be much more flexibly customized. For example, we might want to use longestBatchStep instead of step during the dual averaging.